### DISCUSSION TITLE
Solution to better OSM edits with JOSM presets in quick actions

### DISCUSSION THREAD

**Discussion by laralem** [osmandapp/OsmAnd#11621] [Ideas]
# 🚀 feature request

### Description

Give a bit more controls/features and a **big gift to OpenStreetMap editors** in a single button, the quick actions button: show more buttons in quick actions panel and use the most common/suitable JOSM or iD presets in those buttons.

### Describe the solution you'd like
This is an idea to add at least 1 tag preset (or many presets quickly, see last 3 screenshots) already filled with the most common values. At this moment we can do this manually, introducing every field, but it's a lot of work and a bit limited. To make this worse, OsmAnd doesn't have/recognize every tag, some common, so we have to know them by memory or read the OSM wiki, while editing in the filed sometimes this is not possible.

Also, the worst of all to OSM editors, the quick actions panel shows only a few buttons (6 in each panel), not suitable for many tags used daily by the common OSM user editor. A median/experienced OSM editor would need more or less, lets say 20 panels as seen ant this moment in OsmAnd (120 tags in total, remember, to make quick actions/edits really quickly, for example, I have in my OsmAnd more than one button to add surveillance cameras, 1 button for surveillance cameras in poles in highways to record vehicles, and other button to surveillance cameras in walls to record pedestrians on sidewalks, this is just an example to show we could use/need more than the typical JOSM presets to make really quick edits. So 6 buttons in each panel is not  practical, see solution for this in the mock-ups at the end). We could use some JOSM or iD tag presets to speed this, and adapt if we want some tags to our personal experience/tastes after that.

I know the last screen in the next 3 shown above (JOSM tags presets) has touchable areas a bit too small to fingers, it's just a example mock-up, they should be bigger to avoid mistakes. The option to choose the icon and is color would be a bonus (first and second screen). Some presets will never have a POI type (an house number for example -- addr:housenumber) and in that case OsmAnd shows at this moment a generic icon if we do that. The possibility to add customized icons (in SVG format I think to give the possibility to change the color of them easily), although not very important, it would be a bonus too. The idea to choose the icon and it's color is already available in OsmAnd while choosing a favorite but it's not used in OpenStreetMap POIs/presets/quick actions adding a POI.

In the next first screen we can see the "choose preset" button, the icon and it's color. The icon would be changed immediately by choosing a preset and could be changed after that if needed.

<img src="https://user-images.githubusercontent.com/80917261/117147631-f618e300-ada4-11eb-9494-09aa61060880.png" width="250"/> <img src="https://user-images.githubusercontent.com/80917261/117147745-1648a200-ada5-11eb-8098-917ff86dab70.png" width="250"/> <img src="https://user-images.githubusercontent.com/80917261/117147819-22ccfa80-ada5-11eb-8648-619daed32a45.png" width="250"/> 

If possible, to speed even more this process a lot, instead the option tho choose each tag presets in previous last screen, we could have the possibility to choose all the tags in a single screen we want to use in quick actions (see the check symbol in the last screen next to the tag preset showing the presets chosen) and after this adapt them to our taste if we want (tags used, icon...), maintaining the option to change the icon and the color as shown in previous 3 screens).

<img src="https://user-images.githubusercontent.com/80917261/117148035-5e67c480-ada5-11eb-85a4-c976ef44b6bb.png" width="250"/> <img src="https://user-images.githubusercontent.com/80917261/117148006-5576f300-ada5-11eb-82b3-e2b38c135991.png" width="250"/> <img src="https://user-images.githubusercontent.com/80917261/117148066-658ed280-ada5-11eb-9e8e-6895481a141a.png" width="250"/> 

If needed I can manually create these tags presets in my OsmAnd device (not all JOSM presets are suitable, for example highways presets in OsmAnd) and send the file with that (after a lot of work, but would the result would benefit all users). I couldn't ask a lot more to ease the life of and OSM editor in OsmAnd (edit highways tags would be good also, for example, but this proposal idea would be more important than everything else needed and it's in the actual scope/objectives/capacities of OsmAnd).

### Describe alternatives you've considered

I don't want to ask a lot to OsmAnd developers, the best solution would be the implement all 6 screens mock-ups above or something similar. I've used already common interface elements to be easier for OsmAnd developers.

Some things present here must be refined, like for example in what situations presets groups should show (facilities, amenity, transport...) since at this moment quick actions are not only mean for tags. Could be, if the user add 1 tag preset from a group, that group is shown. The first group not present in these mock-ups could be "Special" with all other actions present at this moment in OsmAnd and not tag related (add favorite, new audio note...)

If this idea is implemented we should consider if OsmAnd should have all or some commons presets in first install, or quick actions should be empty because everyone could add all tags quickly as shown in the last mock-up here.

By order of importance to be implemented, at least to me as and OSM editor:
1- more quick actions buttons in each panel (a must have)
2- organize actions in groups (in this proposal that would be automatic, not giving the option to customize them, but that could be implemented later).
3- add quickly many JOSM presets, and if not possible/accepted, at least give the possibility to add JOSM presets, one at a time
4- customize icons and it's colors
5- organize the order of actions/presets

**Comment by laralem** [comment]
Extra feature: add groups to quick actions panel.
<img src="https://user-images.githubusercontent.com/80917261/117434953-d1e01200-af1c-11eb-915e-b70321747140.png" width="250"/> <img src="https://user-images.githubusercontent.com/80917261/117435055-f0dea400-af1c-11eb-973b-e4dd26c32da4.png" width="250"/> <img src="https://user-images.githubusercontent.com/80917261/117435108-ffc55680-af1c-11eb-9e2a-e16378c00315.png" width="250"/>

Screen 1: activating "Actions group" hides "POI type" and "Add tag" button (see first mock-up in this issue). 
Screen 2: actions groups have a rectangle around icons. 
Screen 3: choosing an actions group while editing the map, is shown a list of refined actions presets according to user preferences. This screen is in a list type and not like the previous screen (icons grid) because the need to show more text describing the specific tags used in that preset.

This suggestion has in mind adding a POI very quickly without the need to edit tags in the field and organize a bit better many actions proposed previously.

**Comment by abdullahO2** [comment]
good idea and if the POI search and filtering was by JOSM presets and it's icons it will be better and easy for users
